### PR TITLE
Fixes Bubblegum

### LINKED ISSUE #67873: bubblegum does not attack
## Testmerges:
The following pull requests are currently test merged:
[#67796: 'Mining Outpost Overhaul: Island Living Edition' by Thunder12345 at commit fc92e70aa9](https://www.github.com/tgstation/tgstation/pull/67796)
[#67669: 'Lemon attempts to optimize lighting [TM]' by LemonInTheDark at commit 9913284f94](https://www.github.com/tgstation/tgstation/pull/67669)
[#65791: 'Optimizes Runechat' by Kylerace at commit 775d002c97](https://www.github.com/tgstation/tgstation/pull/65791)
[#67083: 'Completely removes `proc_holders` from existence. Refactors all wizard, xeno, spider, and genetics powers to be actions. Also refactors and sorts ton of accompanying code. ' by MrMelbert at commit 4219c79a58](https://www.github.com/tgstation/tgstation/pull/67083)

## Reproduction:
shoot bubbles
he never once attempts to charge you just ambling around
occasionally he turns red and walks at you then goes back to ambling
free heck suit

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request

Bubblegum is no longer a pushover. His brain is un-fucked and he learned how to beat your ass again. 

Fixes #67873

## Why It's Good For The Game

Because bubblegum is broken and he's been going through a rough time for almost a month. Bug fixes are good. Broken megafauna are not.

## Changelog

:cl: Nuke
fix: Bubblegum will now beat your ass again.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 1]
```diff
@@ -135,7 +135,7 @@ Difficulty: Hard
 
 /mob/living/simple_animal/hostile/megafauna/bubblegum/proc/get_mobs_on_blood(mob/target)
 	var/list/targets = list(target)
-	. = list()
+	. = list(target)
```
Comment (timothymtorres) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:138]:
```suggestion
	. = list(target) // Bubblegum bubblegum in a dish, how many corpses do you wish
```

**Comment by timothymtorres** [comment]
Bubblegum bubblegum in a dish, how many corpses do you wish?

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm (hunk 1 above)
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:138]: this doesn't look right....

**Comment by ShizCalev** [comment]
#63520 touched all this code
@Whoneedspacee Opinions? 
https://github.com/tgstation/tgstation/pull/63520/files#diff-94d0541458cf8009bd1d509897fac69c8d41d18be1a268f7d04974d121d365c7

**Comment by ShizCalev** [comment]
Ah yes, I see now.

**Comment by ShizCalev** [comment]
I'm gonna go ahead and assume you didn't actually test this since nothing is ever passed to the target argument to start with lol.

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 2]
```diff
@@ -135,7 +135,7 @@ Difficulty: Hard
 
 /mob/living/simple_animal/hostile/megafauna/bubblegum/proc/get_mobs_on_blood(mob/target)
 	var/list/targets = list(target)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:137]:
```suggestion
/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/get_mobs_on_blood()
	var/list/targets = ListTargets()
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm (hunk 2 above)
Comment (StrangeWeirdKitten) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:137]: I did test my stuff. List(target) did make bubblegum's abilities start working.

**Comment by StrangeWeirdKitten** [comment]
I did test my stuff. When I changed the list() into list(targets), bubblegum's full abilities started working again.

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 3]
```diff
@@ -133,9 +133,9 @@ Difficulty: Hard
 	else
 		hallucination_charge_surround.Trigger(target = target)
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/get_mobs_on_blood(mob/target)
-	var/list/targets = list(target)
-	. = list()
+/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/get_mobs_on_blood()
+	var/list/targets = ListTargets()
+	. = list(target)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:138]: can rever this back to . = list()

Comment (StrangeWeirdKitten) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:138]: Should I close the PR then?

**Comment by StrangeWeirdKitten** [comment]
Ok, I made the requested changes. Now the build is failing. 

> I'm gonna go ahead and assume you didn't actually test this as nothing is passed to the target argument (which should have been a list of valid targets anyway) to start with, so this doesn't actually fixing anything since the list is still always empty lol.

List(targets) was working. Bubblegum started charging at targets again. I honestly don't know the fine details of how DM reads its lists, but the list was being defined and properly executed on via code. Does it look wrong? Yeah. Probably. But the changes that was requested and that I pushed into the PR has 14 compile errors now.

**Comment by Fikou** [comment]
the error is on master, not your branch

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**